common/djangoapps/entitlements/models.py

```diff
@@ -0,0 +1,112 @@
+from django.db import models
+from django.contrib.auth.models import User
+from course_modes.models import CourseMode
+
+
+class CourseEntitlement(models.Model):
+    """
+    Represents a Student's Entitlement to a Course Run for a given Course.
+    """
+
+    user = models.ForeignKey(User)
+
+    parent_course_uuid = models.UUIDField(primary_key=True)
+
+    created = models.DateTimeField(auto_now_add=True, db_index=True)
+
+    updated = models.DateTimeField(auto_now=True)
+    # The date that an enrollment must be applied by
+    expiration = models.DateTimeField(null=False)
+
+    # The mode of the Course that will be applied
+    mode = models.CharField(default=CourseMode.DEFAULT_MODE_SLUG, max_length=100)
```

Comment:
Not your fault.  And potentially out of scope for this...  but just wanted to rage a little about continuing to track `mode` as a string

Reply:
So many 👍  💯